pin/gc/gc.go

```diff
@@ -76,14 +76,16 @@ func GC(ctx context.Context, bs bstore.GCBlockstore, dstor dstore.Datastore, pn
 		errors := false
 		var removed uint64
 
+		gchashes := MultihashSetFromCids(gcs)
```

Comment:
Note: we can't just use a multihash set when generating the colored set as we might visit a node with a raw CID first and then *not* visit its children when we visit it using a non-raw CID. Basically, traversal cares about CIDs.